miner/worker_builder.go

```diff
@@ -366,13 +366,25 @@ func (w *worker) simulateBundle(
 		bundleGasFees = new(big.Int)
 	)
 
+	currentState := state.Copy()
+
 	for i, tx := range bundle.Txs {
 		state.SetTxContext(tx.Hash(), i+currentTxCount)
 
+		prevState := currentState.Copy()
```

Comment:
state.Copy is a very performance-intensive action. If you want it to revert to the previous state when a transaction fails, you should use snapshot-related methods. Otherwise, performance will significantly degrade.

Reply:
Fixed. Please review again.